lls/nd.c

```diff
@@ -395,22 +403,23 @@ process_nd_neigh_solicitation(struct lls_config *lls_conf, struct rte_mbuf *buf,
 	}
 
 	/* Make sure buffer is correct size. */
-	RTE_VERIFY(RTE_MBUF_DEFAULT_BUF_SIZE >= ND_NEIGH_PKT_LLADDR_MIN_LEN);
-	if (pkt_len > ND_NEIGH_PKT_LLADDR_MIN_LEN) {
-		if (rte_pktmbuf_trim(buf,
-				pkt_len - ND_NEIGH_PKT_LLADDR_MIN_LEN) < 0) {
+	min_len = ND_NEIGH_PKT_LLADDR_MIN_LEN(l2_len);
+	RTE_VERIFY(RTE_MBUF_DEFAULT_BUF_SIZE >= min_len);
+	if (pkt_len > min_len) {
+		if (rte_pktmbuf_trim(buf, pkt_len - min_len) < 0) {
 			RTE_LOG(ERR, GATEKEEPER, "lls: could not trim packet to correct size for response to a Neighbor Solicitation\n");
 			return -1;
 		}
-	} else if (pkt_len < ND_NEIGH_PKT_LLADDR_MIN_LEN) {
-		if (rte_pktmbuf_append(buf, ND_NEIGH_PKT_LLADDR_MIN_LEN -
-				pkt_len) == NULL) {
+	} else if (pkt_len < min_len) {
+		if (rte_pktmbuf_append(buf, min_len - pkt_len) == NULL) {
 			RTE_LOG(ERR, GATEKEEPER, "lls: could not append space to packet to correct size for response to a Neighbor Solicitation\n");
 			return -1;
 		}
 	}
 
 	if (src_eth_addr != NULL) {
+		int ret;
```

Comment:
Variable `ret` is being used on both branches of this `if` statement, so we can declare it the next bigger scope.